src/utils/bibModelPreprocessing.ts

```diff
@@ -0,0 +1,101 @@
+import type { SearchResult, Note } from "../types/searchTypes"
+
+const getGroupedNotes = (bib: SearchResult) => {
+  const note = bib?.note?.length ? bib.note : null
+  let notesGroupedByNoteType = {}
+
+  /**
+   * getNoteType(note)
+   * Construct label for a note by adding the word 'Note'
+   */
+  const getNoteType = (note: Note) => {
```

Comment:
I think this should be defined outside of this function so it's not instantiated every time it runs.